src/Tests/Microsoft.NET.TestFramework/TestAsset.cs

```diff
@@ -91,6 +97,7 @@ public TestAsset WithSource()
 
         public TestAsset WithTargetFramework(string targetFramework, string projectName = null)
         {
+            TargetFrameworkMoniker = targetFramework;
```

Comment:
Most of our test assets don't set the `TargetFramework` via `WithTargetFramework`.  So this new property would only be set for a small portion of our tests, while it looks like it should work generally.  Can you try to find another way to solve the problem?  For example, you could add a `targetFramewokr` parameter to the `GetExpectedFilesFromBuild` method.